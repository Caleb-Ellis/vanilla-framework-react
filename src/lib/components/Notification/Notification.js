import React from 'react';
import PropTypes from 'prop-types';
import getClassName from '../../utils/getClassName';

const Notification = (props) => {
  const {
    action, children, className, status, caution, information, negative, positive,
  } = props;

  const classNames = getClassName({
    [className]: className,
    'p-notification--information': information,
    'p-notification--positive': positive,
    'p-notification--caution': caution,
    'p-notification--negative': negative,
  }) || 'p-notification';

  return (
    <div className={classNames}>
      <p className="p-notification__response">
        {status && <span className="p-notification__status">{status}</span>}
        {children}
        {action && <a href={action.href} className="p-notification__action">{action.value}</a>}
      </p>
    </div>
  );
};

Notification.defaultProps = {
  action: null,
  caution: false,
  className: undefined,
  information: false,
  negative: false,
  positive: false,
  status: null,
};

Notification.propTypes = {
  action: PropTypes.shape({
    value: PropTypes.string,
    href: PropTypes.string,
  }),
  caution: PropTypes.bool,
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  information: PropTypes.bool,
  negative: PropTypes.bool,
  positive: PropTypes.bool,
  status: PropTypes.string,
};

Notification.displayName = 'Notification';

export default Notification;
