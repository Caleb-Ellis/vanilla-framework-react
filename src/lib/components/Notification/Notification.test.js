import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import Notification from './Notification';

describe('Notification component', () => {
  it('should render a default Notification', () => {
    const notification = ReactTestRenderer.create(
      <Notification>
        This is a default Notification
      </Notification>);
    const json = notification.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a status if prop is provided', () => {
    const notification = ReactTestRenderer.create(
      <Notification status="Status:">
        This is a Notification with a status
      </Notification>);
    const json = notification.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render an action if prop is provided', () => {
    const notification = ReactTestRenderer.create(
      <Notification action={{ value: 'Action', href: '#' }}>
        This is a Notification with an action
      </Notification>);
    const json = notification.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render an information Notification', () => {
    const notification = ReactTestRenderer.create(
      <Notification information>
        This is an information Notification
      </Notification>);
    const json = notification.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a positive Notification', () => {
    const notification = ReactTestRenderer.create(
      <Notification positive>
        This is a positive Notification
      </Notification>);
    const json = notification.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a caution Notification', () => {
    const notification = ReactTestRenderer.create(
      <Notification caution>
        This is a caution Notification
      </Notification>);
    const json = notification.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a negative Notification', () => {
    const notification = ReactTestRenderer.create(
      <Notification negative>
        This is a negative Notification
      </Notification>);
    const json = notification.toJSON();
    expect(json).toMatchSnapshot();
  });
});

