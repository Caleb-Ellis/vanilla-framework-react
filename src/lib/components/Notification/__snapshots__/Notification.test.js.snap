// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification component should render a caution Notification 1`] = `
<div
  className="p-notification--caution"
  style={
    Object {
      "display": "flex",
    }
  }
>
  <p
    className="p-notification__response"
  >
    This is a caution Notification
  </p>
  <button
    aria-label="Close notification"
    className="p-icon--close"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "backgroundSize": "1rem",
        "border": "0",
        "margin": "1.1875rem 1rem auto auto",
        "padding": ".5rem",
      }
    }
  >
    Close
  </button>
</div>
`;

exports[`Notification component should render a default Notification 1`] = `
<div
  className="p-notification"
  style={
    Object {
      "display": "flex",
    }
  }
>
  <p
    className="p-notification__response"
  >
    This is a default Notification
  </p>
  <button
    aria-label="Close notification"
    className="p-icon--close"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "backgroundSize": "1rem",
        "border": "0",
        "margin": "1.1875rem 1rem auto auto",
        "padding": ".5rem",
      }
    }
  >
    Close
  </button>
</div>
`;

exports[`Notification component should render a negative Notification 1`] = `
<div
  className="p-notification--negative"
  style={
    Object {
      "display": "flex",
    }
  }
>
  <p
    className="p-notification__response"
  >
    This is a negative Notification
  </p>
  <button
    aria-label="Close notification"
    className="p-icon--close"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "backgroundSize": "1rem",
        "border": "0",
        "margin": "1.1875rem 1rem auto auto",
        "padding": ".5rem",
      }
    }
  >
    Close
  </button>
</div>
`;

exports[`Notification component should render a positive Notification 1`] = `
<div
  className="p-notification--positive"
  style={
    Object {
      "display": "flex",
    }
  }
>
  <p
    className="p-notification__response"
  >
    This is a positive Notification
  </p>
  <button
    aria-label="Close notification"
    className="p-icon--close"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "backgroundSize": "1rem",
        "border": "0",
        "margin": "1.1875rem 1rem auto auto",
        "padding": ".5rem",
      }
    }
  >
    Close
  </button>
</div>
`;

exports[`Notification component should render a status if prop is provided 1`] = `
<div
  className="p-notification"
  style={
    Object {
      "display": "flex",
    }
  }
>
  <p
    className="p-notification__response"
  >
    <span
      className="p-notification__status"
    >
      Status:
    </span>
    This is a Notification with a status
  </p>
  <button
    aria-label="Close notification"
    className="p-icon--close"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "backgroundSize": "1rem",
        "border": "0",
        "margin": "1.1875rem 1rem auto auto",
        "padding": ".5rem",
      }
    }
  >
    Close
  </button>
</div>
`;

exports[`Notification component should render an action if prop is provided 1`] = `
<div
  className="p-notification"
  style={
    Object {
      "display": "flex",
    }
  }
>
  <p
    className="p-notification__response"
  >
    This is a Notification with an action
    <a
      className="p-notification__action"
      href="#"
    >
      Action
    </a>
  </p>
  <button
    aria-label="Close notification"
    className="p-icon--close"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "backgroundSize": "1rem",
        "border": "0",
        "margin": "1.1875rem 1rem auto auto",
        "padding": ".5rem",
      }
    }
  >
    Close
  </button>
</div>
`;

exports[`Notification component should render an information Notification 1`] = `
<div
  className="p-notification--information"
  style={
    Object {
      "display": "flex",
    }
  }
>
  <p
    className="p-notification__response"
  >
    This is an information Notification
  </p>
  <button
    aria-label="Close notification"
    className="p-icon--close"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "backgroundSize": "1rem",
        "border": "0",
        "margin": "1.1875rem 1rem auto auto",
        "padding": ".5rem",
      }
    }
  >
    Close
  </button>
</div>
`;
